### PR TITLE
Lesson 1 Shmakov

### PR DESCRIPTION
**Pull request by pshmakov**

### CODE REVIEW

File: app/src/main/java/com/csc/lesson1/MainActivity.java [hunk 1]
```diff
@@ -2,11 +2,36 @@
...
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageView;
+
+import com.squareup.picasso.Picasso;
 
 public class MainActivity extends AppCompatActivity {
 
+    private static final String IMAGE_URL = "http://i.imgur.com/DvpvklR.png";
```
Comment (thevery) [inline app/src/main/java/com/csc/lesson1/MainActivity.java:12]: is it really you?

File: app/src/main/java/com/csc/lesson1/MainActivity.java [hunk 2]
```diff
@@ -2,11 +2,36 @@
...
+        // Trying something here...
+        // I want the picture to fit the screen vertically or horizontally
+        // (depending on screen orientation), respecting image proportions
+        imageView.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
+            @Override
+            public void onLayoutChange(View v, int left, int top, int right,
+                                       int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
+                Picasso.with(MainActivity.this)
+                        .load(IMAGE_URL)
+                        .resize(right - left, bottom - top) // one of args is zero, which means use image's aspect ratio
```
Comment (thevery) [inline app/src/main/java/com/csc/lesson1/MainActivity.java:30]: nice!

File: app/src/main/java/com/csc/lesson1/MainActivity.java [hunk 3]
```diff
@@ -2,11 +2,36 @@
...
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout);
+
+        final ImageView imageView = (ImageView) findViewById(R.id.image_view);
+
+        // Trying something here...
+        // I want the picture to fit the screen vertically or horizontally
+        // (depending on screen orientation), respecting image proportions
+        imageView.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
```
Comment (thevery) [inline app/src/main/java/com/csc/lesson1/MainActivity.java:24]: causes extra relayout with layout jump. you should probably use preDrawListener

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,6 @@
 <resources>
     <string name="app_name">lesson1</string>
```
Comment (thevery) [inline app/src/main/res/values/strings.xml:2]: not localized!

**Comment by thevery** [comment]
+:
- Landscape
- Localization
- Caching
- Nice resize!

-:
- Layout jump
- Package is not unique

Total:
10 + 4 - 2 = 12

File: app/src/main/java/com/csc/lesson1/MainActivity.java (hunk 3 above)
Comment (pshmakov) [inline app/src/main/java/com/csc/lesson1/MainActivity.java:24]: But the preDrawListener is called even later than layoutChangeListener!

**Pull request closed without merging**